### PR TITLE
Fix flaky e2e test actor_metadata

### PR DESCRIPTION
**Pull request by MregXN**
# Description
Increase the retry interval and the maximum number of retries to ensure that the test is passed under a greater network delay.

With these modifications, this test has been consistently passing on my own AKS cluster and GitHub Action Workflows.

## Issue reference

Please reference the issue : #6822 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/e2e/actor_metadata/actor_metadata_test.go [hunk 1]
```diff
@@ -225,6 +225,11 @@ func TestActorMetadataEtagRace(t *testing.T) {
 			t.Logf("Sleeping for %d seconds ...", secondsToCheckReminderResult)
 			time.Sleep(secondsToCheckReminderResult * time.Second)
 
+			// Define the backoff strategy
+			bo := backoff.NewExponentialBackOff()
+			bo.InitialInterval = 2 * time.Second
+			var maxRetries uint64 = 20
```
Comment (ItalyPaleAle) [inline tests/e2e/actor_metadata/actor_metadata_test.go:231]: Nit:

```suggestion
			const maxRetries = 20
```

**Review by ItalyPaleAle — DISMISSED** [review]

**Comment by mukundansundar** [comment]
@MregXN the actor metadata tests seems to timeout in windows, can you look into either increasing the timeout for that test alone in windows or changing the retry interval?

**Comment by MregXN** [comment]
> @MregXN the actor metadata tests seems to timeout in windows, can you look into either increasing the timeout for that test alone in windows or changing the retry interval?

Done in [4ec6b12](https://github.com/dapr/dapr/pull/6823/commits/4ec6b127fd3ce9935d766a297e5dd1b6993cebfb).

**Comment by mukundansundar** [comment]
@MregXN why is the timeout increased to 60m? That seems way too long for a test.

**Comment by MregXN** [comment]
> @MregXN why is the timeout increased to 60m? That seems way too long for a test.

It is only for test. Can you trigger it once?  I wanna read more log about this failure.

**Comment by mukundansundar** [comment]
> > @MregXN why is the timeout increased to 60m? That seems way too long for a test.
> 
> It is only for test. Can you trigger it once? I wanna read more log about this failure.

for now just adding the `do-not-merge` label. Will remove it once the PR has been updated.

**Comment by MregXN** [comment]
Hi All. This test can stalbly pass after I roll back the dapt/dapr master branch to commit 2007395c3ad2b8c5 ( which is behind merged PR #6179).

It seems that this PR will lead to failure of test `actor_metadata` but it still get merged. The following are several workflow runs I did in my fork repo.


https://github.com/MregXN/dapr/actions/runs/5915425559
https://github.com/MregXN/dapr/actions/runs/5915425362
https://github.com/MregXN/dapr/actions/runs/5915188208


These runs prove that increasing the number of retries can make `actor_metadata` pass stably when roll back.

**Comment by mukundansundar** [comment]
> st can stalbly pass after I roll back the dap

cc @yaron2 @JoshVanL

**Comment by MregXN** [comment]
> > Hi All. This test can stalbly pass after I roll back the dapt/dapr master branch to commit [2007395](https://github.com/dapr/dapr/commit/2007395c3ad2b8c52ae0a7b64c7811e5830ba10e) ( which is behind merged PR #6179).
> > It seems that PR #6179 will lead to failure of test actor_metadata but it still get merged. The following are several workflow runs I did in my fork repo.
> > https://github.com/MregXN/dapr/actions/runs/5915425559
> > https://github.com/MregXN/dapr/actions/runs/5915425362
> > https://github.com/MregXN/dapr/actions/runs/5915188208
> > These runs prove that increasing the number of retries can make actor_metadata pass stably when roll back.
> > @mukundansundar
> 
> The issue here though is why is it failing in windows alone? And not in linux? If it is identity management overhaul PR as indicated above as `PR 6179` then an investigation is needed.
> 
> cc @yaron2 @JoshVanL

Actually before #6179 is merged the `actor_metadata` failure will occur in both linux and windows. I dont have much idea because  I am not quite familiar with the implementation details of that PR yet.

**Comment by MregXN** [comment]
> Thanks for opening @MregXN. I agree that increasing the test timeout & retries is not great here, unless we know explicitly _why_ this is now needed, and that we accept this is now required when it wasn't before.
> 
> W.r.t 6179, are we sure that this PR is the change which introduced the problem, or could it be later on in the history?

These days I rolled back every commit so that I could locate to this PR. 

Before 6179 is merged, this test may fail in both windows and linux, and generally fails when the partition is set to 5 or higher. [Here](https://github.com/dapr/dapr/actions/runs/5742887338) is a test workflow you could refer
 to.

But When roll back to 6179. The test output log changed. `actor_metadata` test will no longer success, and will report a stable error under windows. At this time, the partition is only set to 2.

**Review by mukundansundar — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**